web-client/src/models/util.ts

```diff
@@ -0,0 +1,5 @@
+/**
+ * Make the properties K optional in type T
+ */
+export type MakePartial<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>> &
```

Comment:
Here's a snippet of code I'll be keeping close by. This is definitely beyond my Typescript prowess xD